README.md

```diff
@@ -246,7 +246,51 @@ makes it very easy to compare edits.
 1. After issuing a PR, other developers can view your edits, add comment and eventually decide whether to pull in your PR.
 1. You can always update your PR by pushing new edits to the same branch under your fork.
 
-# Step 7: Merge after PR is pulled in
+# Step 7: Rebasing 
+
+### Command Line
+
+1. Make sure you've fetched upstream master
+   
+   ```
+   git fetch upstream master
+   ```
+1. Rebase in interactive mode
+
+   ```
+   git rebase upstream/master -i
+   ```
+1. If you have messy commits, flag them with the squash thing. If there's only one
+conflicing commit, just `ctrl-x` out of the interactive window/message. You 
+should see an error message saying that the patch couldn't be applied.
```

Comment:
Maybe more context like?

1. You will be brought into your system editor (depends on your configuration, could be `vim`, `nano` or other ones), with instruction from `git`
```
pick <commit message 1>
pick <commit message 2>
...

# Rebase <your current commit> onto <last commit on upstream/master> (1 command)
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
# d, drop = remove commit
```
As shown in the instruction, you may `squash` a few commits together if you want to keep the history neat. Other options may be used and picking all commits is totally fine too.

Once you are satisfied with the commit history, simply quit your editor (depends on the editor, it is `ctrl+x` in `nano` and `:wq` in `vim`)  and `git` will start rebasing.

If you see `Successfully rebase to...`, congrats, you don't need to do more edits. Just jump to the last step of this section. If you see something like `the patch couldn't be applied..`, then we have to follow the steps below to fix them.